Python version of data-reading

kbrose opened the pull request:
This pull request is still a Work In Progress (WIP). Before it is merged, it needs to build in the new data-streams added by @mmcneill in PR #35.

kbrose committed:
Python data reading framework

The dataframe is a little bit longer than the R version,
will need to figure out why that is.

kbrose committed:
Pandas DF now has same length as R

Have not checked for exact agreement in column
values, but this is a good start

kbrose committed:
Merges in drekdata successfully, simple plot(s)

Implements the R code up to and including the merge of drekdata. Added
some simple plotting functionality for visualization.

kbrose committed:
adding some TODOs, using pandas to plot not mpl

Also set the transparency of the plotting to get a sense of data density.
Honestly cannot tell if there are edge outlines on the circles in the
plot. If there are, I want to turn them off...

tomschenkjr commented:
Ok. Hold-off on accepting then or want it integrated?

kbrose commented:
I'd rather hold off on integrating it. I'm thinking that I can probably hash out the additions within the next day or two, then we can go ahead and merge. That also gives us a little time to wait and see if any one else has troubles running this.

tomschenkjr commented:
10-4.


Tom Schenk Jr.

Chief Data Officer

Department of Innovation and Technology

City of Chicago

(312) 744-2770

tom.schenk@cityofchicago.org

data.cityofchicago.org


________________________________
From: Kevin Rose <notifications@github.com>
Sent: Tuesday, January 19, 2016 11:49 PM
To: Chicago/e-coli-beach-predictions
Cc: Schenk, Tom
Subject: Re: [e-coli-beach-predictions] WIP: Python version of data-reading (#36)


I'd rather hold off on integrating it. I'm thinking that I can probably hash out the additions within the next day or two, then we can go ahead and merge. That also gives us a little time to wait and see if any one else has troubles running this.

-
Reply to this email directly or view it on GitHub<https://github.com/Chicago/e-coli-beach-predictions/pull/36#issuecomment-173095165>.

[https://avatars2.githubusercontent.com/u/4053170?v=3&s=400]<https://github.com/Chicago/e-coli-beach-predictions/pull/36#issuecomment-173095165>

WIP: Python version of data-reading by kbrose · Pull Request #36 · Chicago/e-coli-beach-predictions<https://github.com/Chicago/e-coli-beach-predictions/pull/36#issuecomment-173095165>
github.com
This pull request is still a Work In Progress (WIP). Before it is merged, it needs to build in the new data-streams added by @mmcneill in PR #35.




________________________________
This e-mail, and any attachments thereto, is intended only for use by the addressee(s) named herein and may contain legally privileged and/or confidential information. If you are not the intended recipient of this e-mail (or the person responsible for delivering this document to the intended recipient), you are hereby notified that any dissemination, distribution, printing or copying of this e-mail, and any attachment thereto, is strictly prohibited. If you have received this e-mail in error, please respond to the individual sending the message, and permanently delete the original and any copy of any e-mail and printout thereof.

kbrose committed:
Adding some PEP8 compliance

read_data.py brought into better PEP8 compliance, except for E231
requiring extra space around some delimiters such as ":", which doesn't
really make sense as a requirement for matrix slicing in my opinion.

PEP8 compliance did not come sooner b/c my linter was broken.

kbrose committed:
Removing plotting commands

They don't have a place in the data-reading file. Will push those methods
to a separate file specifically for visualizations.

kbrose committed:
adding basic logging

Has already helped identify many cases where our manual re-casting of the
column names was incorrect - seeing things like "Montrose" in the
"Reading.1" column and whatnot. Will need to investigate further.

kbrose committed:
Adding sensor data

The sensor data is pivoted to ensure that each beach-day has exactly
one row in the final data-frame. This results in a lot of columns (240
right now), but I think it is ultimately necessary. Holiday data is
loaded, but has not been merged.

kbrose committed:
Fixed bug with merge removing lots of rows

Pandas was doing an implicit inner join instead of an outer join, so
merging in the drek data and weater/water station data was removing any
row from before 2015.

kbrose committed:
Adding beach lat/long data

Along with a quick function to read it in.

kbrose committed:
Adding visualizations file

Will include more advanced visualization functions. Also updated
.gitignore file to ignore python binaries.
